src/visualization/types/Mosaic/options.tsx

```diff
@@ -97,32 +154,25 @@ const MosaicOptions: FC<Props> = ({properties, results, update}) => {
           </Form.Element>
           <Form.Element label="X Column">
             <SelectDropdown
-              options={numericColumns}
+              options={xDataColumn}
               selectedOption={xColumn || 'Build a query before selecting...'}
               onSelect={xColumn => {
                 update({xColumn})
               }}
               testID="dropdown-x"
               buttonStatus={
-                numericColumns.length == 0
+                xDataColumn.length == 0
```

Comment:
`.length` will always return a number, so we can do a strict check here `===` since `==` at the component level often implies that number and string values are going to be interchanged

Reply:
Done.